dataprofiler/tests/profilers/profiler_options/test_categorical_options.py

```diff
@@ -55,24 +108,115 @@ def _assert_set_helper(prop, value):
+        )
+        options = self.get_options(
+            max_sample_size_to_check_stop_condition=2.0,
+            stop_condition_unique_value_ratio=2.0,
+        )
+        self.assertEqual(
+            [expected_max_sample_size_type_error], options._validate_helper()
+        )
+
+        # These ones should throw a max_sample_size_type_error
+        expected_unique_value_ratio_type_error = (
+            "{}.stop_condition_unique_value_ratio must be either None"
+            " or a non-negative float".format(optpth)
+        )
+
+        options = self.get_options(
+            stop_condition_unique_value_ratio=2,
+            max_sample_size_to_check_stop_condition=20,
+        )
+        self.assertEqual(
+            [expected_unique_value_ratio_type_error], options._validate_helper()
+        )
+        options = self.get_options(
+            stop_condition_unique_value_ratio="E",
+            max_sample_size_to_check_stop_condition=20,
+        )
+        self.assertEqual(
+            [expected_unique_value_ratio_type_error], options._validate_helper()
+        )
+        options = self.get_options(
+            stop_condition_unique_value_ratio=-2.0,
+            max_sample_size_to_check_stop_condition=20,
+        )
+        self.assertEqual(
+            [expected_unique_value_ratio_type_error], options._validate_helper()
+        )
+
+        # These ones should throw a stop_condition_error
+        expected_stop_condition_error = (
+            "Both, {}.max_sample_size_to_check_stop_condition and "
+            "{}.stop_condition_unique_value_ratio, options either need to be "
+            "set or not set.".format(optpth, optpth)
+        )
+
+        options = self.get_options(max_sample_size_to_check_stop_condition=20)
+        self.assertEqual([expected_stop_condition_error], options._validate_helper())
+        options = self.get_options(stop_condition_unique_value_ratio=2.0)
+        self.assertEqual([expected_stop_condition_error], options._validate_helper())
+        options = self.get_options(
+            max_sample_size_to_check_stop_condition=20,
+            stop_condition_unique_value_ratio=None,
+        )
+        self.assertEqual([expected_stop_condition_error], options._validate_helper())
+
         # These ones should not
-        expected_error = ""
         options = self.get_options(top_k_categories=6)
         self.assertEqual([], options._validate_helper())
         options = self.get_options(top_k_categories=None)
         self.assertEqual([], options._validate_helper())
 
+        options = self.get_options(
+            max_sample_size_to_check_stop_condition=20,
+            stop_condition_unique_value_ratio=2.0,
+        )
+        self.assertEqual([], options._validate_helper())
+        options = self.get_options(
+            max_sample_size_to_check_stop_condition=0,
+            stop_condition_unique_value_ratio=0.0,
+        )
```

Comment:
so the condition we discussed before, it might be good to make one of these a value. and one 0 to test that condition.

Reply:
could do this as 
`stop_condition_unique_value_ratio=0.5,`

and in check above
set `stop_condition_unique_value_ratio=0.0`

so we don't need more tests to do the same checks